src/test/java/com/google/research/bleth/simulator/TracingSimulationTest.java

```diff
@@ -0,0 +1,374 @@
+        observers.add(observer2);
+
+        Simulation.SimulationBuilderFromExisting builder = new TracingSimulation.TracingSimulationBuilderFromExisting()
+                .setId(SIMULATION_ID)
+                .setMaxNumberOfRounds(MAX_ROUNDS_EQUALS_TWO)
+                .setBeaconMovementStrategy(MOVE_UP)
+                .setObserverMovementStrategy(STATIONARY)
+                .setAwakenessStrategyFactory(DUMMY_AWAKENESS_STRATEGY_FACTORY)
+                .setRadius(RADIUS_EQUALS_ONE)
+                .setRealBoard(board)
+                .setBeacons(beacons)
+                .setObservers(observers);
+
+        assertThrows(NullPointerException.class, builder::build);
+    }
+
+    @Test
+    public void initSimulationFromExistingWithoutAgentsShouldThrowException() {
+        Board board = new Board(BOARD_DIMENSION_EQUALS_TWO, BOARD_DIMENSION_EQUALS_TWO);
+
+        Simulation.SimulationBuilderFromExisting builder = new TracingSimulation.TracingSimulationBuilderFromExisting()
+                .setId(SIMULATION_ID)
+                .setMaxNumberOfRounds(MAX_ROUNDS_EQUALS_TWO)
+                .setBeaconMovementStrategy(MOVE_UP)
+                .setObserverMovementStrategy(STATIONARY)
+                .setAwakenessStrategyFactory(DUMMY_AWAKENESS_STRATEGY_FACTORY)
+                .setRadius(RADIUS_EQUALS_ONE)
+                .setRealBoard(board)
+                .setResolver(resolver);
+
+        assertThrows(NullPointerException.class, builder::build);
+    }
+
+    @Test
+    public void initSimulationFromExistingWithoutBoardShouldThrowException() {
+        ArrayList<Beacon> beacons = new ArrayList<>();
+        beacons.add(beacon1);
+        beacons.add(beacon2);
+        ArrayList<Observer> observers = new ArrayList<>();
+        observers.add(observer1);
+        observers.add(observer2);
+
+        Simulation.SimulationBuilderFromExisting builder = new TracingSimulation.TracingSimulationBuilderFromExisting()
+                .setId(SIMULATION_ID)
+                .setMaxNumberOfRounds(MAX_ROUNDS_EQUALS_TWO)
+                .setBeaconMovementStrategy(MOVE_UP)
+                .setObserverMovementStrategy(STATIONARY)
+                .setAwakenessStrategyFactory(DUMMY_AWAKENESS_STRATEGY_FACTORY)
+                .setRadius(RADIUS_EQUALS_ONE)
+                .setResolver(resolver)
+                .setBeacons(beacons)
+                .setObservers(observers);
+
+        assertThrows(NullPointerException.class, builder::build);
+    }
+
+    // Helper classes.
+
+    private static class moveUp implements IMovementStrategy {
+
+        @Override
+        public Location moveTo(Board board, Location currentLocation) {
+            Location newLocation = currentLocation.moveInDirection(Direction.UP);
+            if (board.isLocationValid(newLocation)) {
+                return newLocation;
+            }
+            return currentLocation;
+        }
+    }
+
+    private static class FakeResolver implements IResolver {
```

Comment:
And again:
"Consider splitting this to a separate file (even if it's not going to be reused - just so it doesn't clutter the tests file)".
(Unless you're waiting for the FakeResolver I wrote.)